packages/idempotency/src/IdempotencyHandler.ts

```diff
@@ -313,11 +314,17 @@ export class IdempotencyHandler<Func extends AnyFunction> {
         );
       } catch (e) {
         if (e instanceof IdempotencyItemAlreadyExistsError) {
-          const idempotencyRecord: IdempotencyRecord =
-            e.existingRecord ||
-            (await this.#persistenceStore.getRecord(
+          let idempotencyRecord = e.existingRecord;
+          if (idempotencyRecord !== undefined) {
+            this.#persistenceStore.validateExistingRecord(
+              idempotencyRecord,
               this.#functionPayloadToBeHashed
-            ));
+            );
+          } else {
```

Comment:
to help your future self, add a comment on the else block. For anyone glancing through quickly, it might read as validation only happens if an existing record exists, but [jmespath validation also occurs on get after cache retrieval](https://github.com/aws-powertools/powertools-lambda-typescript/blob/5e336d42f30ba30a0a3f44c67c7626c9aae6af60/packages/idempotency/src/persistence/BasePersistenceLayer.ts#L110).

As I read a third time, it's almost as if the private method is `processExistingRecord`, which then calls `validatePayload`, followed by `saveToCache`.

Reply:
Good point, I have added comments on both branches to make it clearer.